system/lib/emmalloc.c

```diff
@@ -791,6 +789,13 @@ static void *allocate_memory(size_t alignment, size_t size)
 
   // We were unable to find a free memory region. Must sbrk() in more memory!
   size_t numBytesToClaim = size+sizeof(Region)*3;
+  // Take into account the alignment as well. For typical alignment we don't
+  // need to add anything here (so we do nothing if the alignment is equal to
+  // MALLOC_ALIGNMENT), but it can matter if the alignment is very high. In that
+  // case, not adding the alignment can lead to this sbrk not giving us enough
+  // (in which case, the next attempt fails and will sbrk the same amount again,
+  // potentially allocating a lot more memory than necessary).
```

Comment:
I'm not sure I understand the status quo.. does did this work at all without this change?

Reply:
It worked, but it would do the recursion in the earlier code twice. So it allocated a lot more than it needed, until it finally found enough room for the right size with the right allocation.

Reply:
Wouldn't iteratively doing this actually use less memory in some cases?  i.e. you could get lucky and get a good alignmenty without quite as much over allocation? 

With this change are are guaranteed to waste alignment bytes on each allocation aren't we?  In the mimalloc case that would be ~4Mb for each allocation.   Or we we return the over-allocated chunk to the free list?

Reply:
I think this we should wait for @juj to weigh in there.

Reply:
Talking offline, @juj might not be available to review this for a bit and suggested it land for now if it is urgent, and be reviewed later. I think that makes sense as I'd like to get this in to avoid mimalloc using 2x the memory it needs to, in the first release with mimalloc. @sbc100 does that sound ok?